site/docs/relations/logical_relations.md

```diff
@@ -19,11 +19,15 @@ The read operator is an operator that produces one output. A simple example woul
 | ----------------- | ------------------------------------------------------------ | ------------------------------------ |
 | Definition        | The contents of the read property definition.                | Required                             |
 | Direct Schema     | Defines the schema of the output of the read (before any projection or emit remapping/hiding). | Required                             |
-| Filter            | A boolean Substrait expression that describes the filter of an iceberg dataset. TBD: define how field referencing works. | Optional, defaults to none.          |
+| Filter            | A boolean Substrait expression that describes a filter that must be applied to the data. The filter should be interpreted against the direct schema and not the masked projected schema. As a result fields that are used for filtering do not have to be included in the output. | Optional, defaults to none.          |
 | Projection        | A masked complex expression describing the portions of the content that should be read | Optional, defaults to all of schema  |
 | Output properties | Declaration of orderedness and/or distribution properties this read produces. | Optional, defaults to no properties. |
 | Properties        | A list of name/value pairs associated with the read.         | Optional, defaults to empty          |
 
+### Read Filtering
+
+Consumers can often take advantage of a ReadRel's filter property, combined with file metadata, statistics, and indices to reduce the amount of data the needs to be read.  This technique is often referred to as "pushdown filtering".  In many cases this is inexact and a filter can only be partially satisfied.  In these cases the consumer must apply some kind of in-memory filtering operation to fully satisfy the filter.
```

Comment:
I think we need this to have a boolean called filter_best_effort which defaults to false (and is therefore backwards compatible with the current definition/intention). If set to true, the scanner may only do best effort filtering. As you put it right now, it sounds like the correct interpretation of a filter is best effort (e.g. may be completed ignored). That wasn't the intention and doesn't feel consistent with least surprise.

Reply:
I thought the plan was to add such a field later.  But I don't mind adding one now.

> As you put it right now, it sounds like the correct interpretation of a filter is best effort

I need to work on my phrasing then.  I was trying to imply the exact opposite.

Reply:
Should this be a boolean options or an enum with a 2 values?  I can't think of anything between "best_effort" and "exact" at the moment but maybe someone would want something where a certain class of expressions must be satisfied? (e.g. "equalities" to mean that "x == 0 and is_cool(y)" should at least evaluate "x == 0"?)

Reply:
Lol. If you meant the opposite, I'm fine without the field. I also think it is fine to add the field as this definitely is fairly common.

Reply:
I suppose an alternate approach per your comments is to add a second field that is best_effort_filter, which if populated, may be applied. This allows people to separate the two if they want. (your in-between case)

Reply:
I've updated my wording to be more clear.  I think it's best to leave the `best_effort_filter` out at the moment.  I'm not sure if this is just laziness on my part but my rationalization is that this will present producers with a difficult dilemma picking which one to use and we should wait until there is a concrete case so we can better understand.